test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs

```diff
@@ -622,6 +623,39 @@ public HubConnectionTests(ServerFixture<Startup> serverFixture, ITestOutputHelpe
             }
         }
 
+        [Fact]
+        public async Task WebSocketOptionsAreApplied()
+        {
+            using (StartLog(out var loggerFactory))
+            {
+                // System.Net has a TransportType type which means we need to fully-qualify this rather than 'use' the namespace
+                var cookieJar = new System.Net.CookieContainer();
```

Comment:
I'd like to see the usage with the HubConnectionBuilder. We might need to add an extensionmethod to make this not terrible.